examples/seismic/tutorials/05_dask.ipynb

```diff
@@ -0,0 +1,496 @@
+    "\n",
+    "This tutorial uses the same synthetic datasets and model setup as the previous two tutorals, so check back if you get lost on parts of the code specific to Devito or SciPy.optimize. "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## What is Dask?\n",
+    "\n",
+    "> [Dask](https://dask.pydata.org/en/latest/#dask) is a flexible parallel computing library for analytic computing.\n",
+    ">\n",
+    "> Dask is composed of two components:\n",
+    ">\n",
+    "> * Dynamic task scheduling optimized for computation...\n",
+    "> * “Big Data” collections like parallel arrays, dataframes, and lists that extend common interfaces like NumPy, Pandas, or Python iterators to larger-than-memory or distributed environments. These parallel collections run on top of the dynamic task schedulers.\n",
+    ">\n",
+    "> Dask emphasizes the following virtues:\n",
+    "> \n",
+    "> * Familiar: Provides parallelized NumPy array and Pandas DataFrame objects\n",
+    "> * Flexible: Provides a task scheduling interface for more custom workloads and integration with other projects.\n",
+    "> * Native: Enables distributed computing in Pure Python with access to the PyData stack.\n",
+    "> * Fast: Operates with low overhead, low latency, and minimal serialization necessary for fast numerical algorithms\n",
+    "> * Scales up: Runs resiliently on clusters with 1000s of cores\n",
+    "> * Scales down: Trivial to set up and run on a laptop in a single process\n",
+    "> * Responsive: Designed with interactive computing in mind it provides rapid feedback and diagnostics to aid humans\n",
+    "\n",
+    "**We are going to use it here to parallelise the computation of the functional and gradient as this is the vast bulk of the computational expense of FWI and it is trivially parallel over data shots.**"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Setting up (synthetic) data\n",
+    "We are going to set up the same synthetic test case as for the previous tutorial (refer back for details). The code below is slightly re-engineered to make it suitable for using with scipy.optimize.minimize."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "#NBVAL_IGNORE_OUTPUT\n",
+    "from examples.seismic import Model, demo_model\n",
+    "\n",
+    "# Define the grid parameters\n",
+    "def get_grid():\n",
+    "    shape = (101, 101)    # Number of grid point (nx, nz)\n",
+    "    spacing = (10., 10.)  # Grid spacing in m. The domain size is now 1km by 1km\n",
+    "    origin = (0., 0.)     # Need origin to define relative source and receiver locations\n",
+    "\n",
+    "    return shape, spacing, origin\n",
+    "\n",
+    "# Define the test phantom; in this case we are using a simple circle\n",
```

Comment:
"test phantom" does not make sense to me. Is this a geophysics term?

Reply:
https://www.google.co.uk/search?ei=QtQCWvTJFonKsAej-Jq4Bg&q=test+phantom&oq=test+phantom&gs_l=psy-ab.3..0i67k1j0j0i22i30k1l8.1558.1753.0.1988.2.2.0.0.0.0.176.176.0j1.1.0....0...1.1.64.psy-ab..1.1.175....0.IY3c6AsdNLs